File: pkg/plugin/builtin/example_plugin/scripts/handler.ts
function handler(): void {
    const input = $args.input || "world";
    $logger.info("Processing input: " + input);

    const result = {
        message: "Hello, " + input + "!",
        timestamp: new Date().toISOString(),
        limits: $limits,
    };

    $done({ success: true, data: result });
}

handler();

File: pkg/plugin/builtin/lua_demo_plugin/scripts/handler.lua
-- Lua Demo Plugin
--
-- Echos back a greeting with the provided name.
-- Demonstrates basic Lua plugin structure: handle() function + done() callback.

function handle(args)
    local name = args["name"] or "world"

    logger:info("Greeting " .. name)

    done({
        success = true,
        data = {
            message = "Hello from Lua, " .. name .. "!",
            plugin_name = plugin_name,
            name_length = string.len(name)
        }
    })
end
